frontend/web/components/release-pipelines/CreatePipelineStage.tsx

```diff
@@ -134,7 +148,23 @@ const CreatePipelineStage = ({
 
     const action_body = { enabled }
 
-    handleOnChange('actions', [{ action_body, action_type }])
+    const actions = stageData.actions.map((action, idx) => {
+      if (idx === actionIndex) {
+        return { action_body, action_type }
+      }
+      return action
+    })
+    handleOnChange('actions', actions)
+  }
+
+  const handleAddAction = () => {
+    const defaultDraftAction = NEW_PIPELINE_STAGE.actions[0]
```

Comment:
Being a bit picky we could create `DEFAULT_ACTION` and `DEFAULT_TRIGGER` as constants to build the NEW_PIPELINE_STAGE and use it here and in CreateReleasePipeline directly